analyzer/src/lib.rs

```diff
@@ -28,20 +28,15 @@ pub fn analyze(module: &fe::Module, file_id: SourceFileId) -> Result<Context, An
             if context.diagnostics.is_empty() {
                 Ok(context)
             } else {
-                Err(AnalyzerError {
-                    diagnostics: context.diagnostics,
-                    classic: None,
-                })
+                Err(AnalyzerError(context.diagnostics))
             }
         }
-        Err(err) => Err(AnalyzerError {
-            diagnostics: context.diagnostics,
-            classic: if err.kind == ErrorKind::Fatal {
-                None
-            } else {
-                Some(err)
-            },
-        }),
+        Err(FatalError) => {
+            if context.diagnostics.is_empty() {
+                panic!("Expected at least one error")
```

Comment:
This is a safety catch for us. If we returned a `FatalError` but the `diagnostics` are empty we've obviously forgot to report an error somewhere.